include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -917,8 +917,8 @@ namespace VectorTools
       additional_data.mapping_update_flags = (update_values | update_JxW_values);
       MatrixFree<dim, Number> matrix_free;
       matrix_free.reinit (mapping, dof, constraints,
-                          QGauss<1>(fe_degree+1), additional_data);
-      typedef MatrixFreeOperators::MassOperator<dim, fe_degree, 1, Number> MatrixType;
+                          QGauss<1>(fe_degree+2), additional_data);
+      typedef MatrixFreeOperators::MassOperator<dim, fe_degree, fe_degree+2, 1, Number> MatrixType;
```

Comment:
Here is the only complaint I would have: Why do you hardcode `degree+2` at this point? This is going to be a bit slower and as you said in #3262, one is fine with `degree+1` in more than 90% of times. Shouldn't you put an additional template argument (increasing compilation time, hah), and step into the variants `fe_degree+1` and `fe_degree+2` in the `project function below?

(Sorry for being slow, but that's what happens when one starts taking some administrative tasks...)